samza-core/src/main/java/org/apache/samza/metrics/reporter/Metrics.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.metrics.reporter;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+
+public class Metrics {
+  private final Map<String, Map<String, Object>> immutableMetrics = new HashMap<>();
+
+  public Metrics() {
+    this(Collections.emptyMap());
+  }
+
+  public Metrics(Map<String, Map<String, Object>> metrics) {
+    metrics.forEach((groupKey, groupValue) -> {
+      Map<String, Object> immutableMetricGroup = new HashMap<>();
+      groupValue.forEach(immutableMetricGroup::put);
```

Comment:
Is this line `immutableMetricGroup.putAll(groupValue)`? Or just `Map<String, Object> immutableMetricGroup = groupValue.clone()`?

Don't think this is a deep clone either way.

Reply:
Ah good call, there is a more concise way to do this. Updated.